### PR TITLE
Use failure for toxencryptsave

### PR DESCRIPTION
**Pull request by kurnevsky**

### CODE REVIEW

File: src/toxencryptsave/mod.rs [hunk 1]
```diff
@@ -167,7 +165,7 @@ impl PassKey {
                               // ↓ don't
     let passkey = PassKey::new(&[0]).expect("Failed to unwrap PassKey!");
 
-    assert_eq!(Err(EncryptionError::Null), passkey.encrypt(&[]));
+    assert_eq!(passkey.encrypt(&[]), Err(EncryptionError::Null));
```
Comment (kpp) [inline src/toxencryptsave/mod.rs]: Why did you change the order of rhs & lhs?

Comment (kurnevsky) [reply src/toxencryptsave/mod.rs]: We mostly use this ordering. Also it used to be in rust src: https://github.com/rust-lang/rust/pull/19647/files
But can revert it - it's not too important.

Comment (kpp) [reply src/toxencryptsave/mod.rs]: Right!
